sdks/go/pkg/beam/core/util/reflectx/functions.go

```diff
@@ -40,6 +40,8 @@ func FunctionName(fn interface{}) string {
 // points to a valid function implementation.
 func LoadFunction(ptr uintptr, t reflect.Type) interface{} {
 	v := reflect.New(t).Elem()
-	*(*uintptr)(unsafe.Pointer(v.Addr().Pointer())) = (uintptr)(unsafe.Pointer(&ptr))
+	p := new(uintptr)
+	*p = ptr
+	*(*unsafe.Pointer)(unsafe.Pointer(v.Addr().Pointer())) = unsafe.Pointer(p)
```

Comment:
I had to read up a bit on how reflect and unsafe work to understand what was happening here, but I still don't understand exactly why the previous version was invalid and this one valid. (I'm using https://golang.org/pkg/unsafe/#Pointer as a guideline).

Is it because in the old version we simply put in the address to `ptr`, and in the new one we actually create a new copy?

Or is it because in the old version we're reassigning to `v` as a `uintptr` instead of as an `unsafe.Pointer`?

Those are the only real differences I could figure out.

Reply:
The problem was that the original code relies on the behavior that when &ptr is evaluated, the resulting anonymous value will be on the heap. The new compiler is able to not put it on the heap, and does so, which causes this code to crash in Go 1.13.

This change makes the expected behavior explicit by allocating p on the heap and then assigning it's pointed-to value to ptr, thus p == &ptr and the remaining code works as before.

The unit test exercises this function. I verified that the old code crashed under Go 1.13 (worked in Go 1.12) and the new code works in both versions.

Reply:
Thanks for the nice summary!